cdap-client/src/main/java/io/cdap/cdap/client/ApplicationClient.java

```diff
@@ -234,13 +234,23 @@ public ApplicationDetail get(ApplicationId appId)
    * Upgrades an application.
    *
    * @param app the application to delete
+   * @param artifactScopes artifact scopes to consider while searching for latest artifacts.
+   * @param allowSnapshot should consider snapshot artifacts or not.
    * @throws ApplicationNotFoundException if the application with the given ID was not found
    * @throws IOException if a network error occurred
    * @throws UnauthenticatedException if the request is not authorized successfully in the gateway server
    */
-  public void upgradeApplication(ApplicationId app)
+  public void upgradeApplication(ApplicationId app, @Nullable Set<String> artifactScopes, boolean allowSnapshot)
```

Comment:
Instead of modify an existing method (this is public API), add overloaded method that take extra parameters and have the original method calls the newly added one.

Also, use empty set instead of `null` for collection.

Reply:
the artifactSCopes shouldn't be null